src/cpp/polars/SeriesMask.cpp

```diff
@@ -21,6 +21,27 @@ namespace polars {
         //assert(!arma::any(v > 1));  // Np SeriesMask values may be greater than 1
     };
 
+    SeriesMask SeriesMask::iloc(int from, int to, int step) const {
+
+        if(empty()){
+            return SeriesMask();
+        }
+
+        arma::uvec pos;
+
+        if((from < 0) && (to > 0)){
+            auto effective_from =  values().size() + from;
+            pos = arma::regspace<arma::uvec>(effective_from, step, to-1);
+        }
+        else if(from < 0 && to < 0){
+            pos = arma::regspace<arma::uvec>(values().size() + from, step, values().size()+to-1);
+        }
+        else {
+            pos = arma::regspace<arma::uvec>(from,  step,  to-1);
+        }
```

Comment:
This section is kind of hard to reason about. Can you deal with determining the effective from separately? Something like:
```c++
auto effective_from - from;
if (from < 0) {
  effective_from = from + size();
}

//... ditto effective_to

pos = arma::regspace<arma::uvec>(effective_from, step, effective_to-1);
```

Reply:
This has been now addressed so resolving.